include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -179,16 +179,16 @@ __serial_merge(const _Rng1& __rng1, const _Rng2& __rng2, _Rng3& __rng3, const _I
         if constexpr (__can_use_ternary_op_v<decltype(__rng1[__rng1_idx]), decltype(__rng2[__rng2_idx])>)
         {
             // This implementation is required for performance optimization
-            __rng3[__rng3_idx] = (!__rng1_idx_less_n1 || __rng1_idx_less_n1 && __rng2_idx_less_n2 &&
-                                                             __comp(__rng2[__rng2_idx], __rng1[__rng1_idx]))
+            __rng3[__rng3_idx] = (!__rng1_idx_less_n1 || (__rng1_idx_less_n1 && __rng2_idx_less_n2 &&
+                                                          __comp(__rng2[__rng2_idx], __rng1[__rng1_idx])))
```

Comment:
Why the redundant parentheses were added?

Reply:
See the answer below